### PR TITLE
sdk regenerated - version v3.0.0-beta

### PR DESCRIPTION
**Pull request by garimakhulbe**
Major changes is this release..
* APIs are updated to latest versions mentioned in CHANGELOG.md.
* ApiVersion and BaseURI parameters are removed from method calls.
* go-autorest related changes.

### CODE REVIEW

**Comment by garimakhulbe** [comment]
@ahmetalpbalkan @brendandixon please take a look.

**Comment by colemickens** [comment]
>BaseURI parameters are removed from method calls and not exposed to users now

Why? Is there another way to use the SDK against non-public-cloud endpoints?

File: arm/dns/recordsets.go [hunk 1]
```diff
@@ -253,71 +263,64 @@ func (client RecordSetsClient) GetResponder(resp *http.Response) (result RecordS
 	return
 }
 
-// List lists the RecordSets of a specified type in a DNS zone.
+// ListAllInResourceGroup lists all RecordSets in a DNS zone.
```
Comment (colemickens) [inline arm/dns/recordsets.go]: This comment and function name don't seem to align.

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,51 @@
...
+## `v3.0.0-beta`
+
+This release brings the Go SDK ARM packages up-to-date with Azure ARM Swagger files for most
+services. Since the underlying [Swagger files](https://github.com/Azure/azure-rest-api-specs)
+continue to change substantially, the ARM packages are still in *beta* status.
+
+The ARM packages now align with the following API versions (*highlighted* packages are new or
+updated in this release):
+
+- arm/authorization 2015-07-01
```
Comment (ahmetalpbalkan) [inline CHANGELOG.md]: fyi this is a markdown file. you can use tables, code blocks etc.

File: CHANGELOG.md [hunk 3]
```diff
@@ -2,6 +2,51 @@
...
+- *arm/logic 2015-08-01-preview*
+- *arm/network 2016-03-30*
+- *arm/redis 2016-04-01*
+- *arm/resources/resources 2016-02-01*
+- *arm/resources/policy 2015-10-01-preview*
+- *arm/resources/locks 2015-01-01* (earlier resources/authorization)
+- *arm/scheduler 2016-03-01*
+- *arm/storage 2016-01-01*
+- *arm/search 2015-02-28* 
+<br/>*----------new------------*
```
Comment (ahmetalpbalkan) [inline CHANGELOG.md]: no need for this br. just give an empty line.

**Comment by ahmetalpbalkan** [comment]
LGTM except markdown issues. If we're gonna add a table it should probably be in a more readable format, such as tabular:

| api | version | note |
|----|-----|----|
| arm/search | 2016-02-28 | new |
| arm/cdn | 2016-04-02 | updated |

**Comment by garimakhulbe** [comment]
@colemickens Yes, they are still exported. You can directly update them via client.

File: arm/dns/recordsets.go (hunk 1 above)
Comment (garimakhulbe) [inline arm/dns/recordsets.go]: @colemickens Coming from swagger, can't update and it seems fine I guess.  
```Go
// ListAllInResourceGroup lists all RecordSets in a DNS zone. (method description)
//
// resourceGroupName is the name of the resource group that contains the zone. (argument description)
// zoneName is the name of the zone from which to enumerate RecordSets. top
// is query parameters. If null is passed returns the default number of
// zones.
func (client RecordSetsClient) ListAllInResourceGroup(resourceGroupName string, zoneName string, top string) (result RecordSetListResult, err error) {
	req, err := client.ListAllInResourceGroupPreparer(resourceGroupName, zoneName, top)
	if err != nil {
		return result, autorest.N
```

**Comment by ahmetalpbalkan** [comment]
@garimakhulbe great, thanks. it would be nice to add the changelog to release notes from now on.

**Pull request merged**